man/rmdhunks/vcr-design.Rmd

```diff
@@ -0,0 +1,152 @@
+intended for general use. Whenever you make calls to `vcr_configure()`
+or other configuration functions, `vcr_c` is affected.
+
+### Cassette class
+
+`Cassette` is an R6 class that handles internals/state for each cassette.
+Each time you run `use_cassette()` this class is used. The class has quite
+a few methods in it, so there's a lot going on in the class. Ideally
+the class would be separated into subclasses to handle similar sets
+of logic, but there's not an easy way to do that with R6. 
+
+Of note in `Cassette` is that when called, within the `initialize()` 
+call `webmockr` is used to create webmockr stubs. 
+
+### How HTTP requests are handled
+
+Within `webmockr`, there's calls to the vcr class `RequestHandler`, which
+has child classes `RequestHandlerCrul` and `RequestHandlerHttr` for 
+`crul` and `httr`, respectively. These classes determine what to do with
+each HTTP request. The options for each HTTP request include:
+
+- **Ignored** You can ignore HTTP requests under certain rules using the 
+configuration options `ignore_hosts` and `ignore_localhost`
+- **Stubbed by vcr** This is an HTTP request for which a match is found
+in the cassette defined in the `use_cassette()`/`insert_cassette()` call.
+In this case the matching request/response from the cassette is returned
+with no real HTTP request allowed.
+- **Recordable** This is an HTTP request for which no match is found
+in the cassette defined in the `use_cassette()`/`insert_cassette()` call.
+In this case a real HTTP request is allowed, and the request/response is
+recorded to the cassette.
+- **Unhandled** This is a group of cases, all of which cause an error
+to be thrown with a message trying to help the user figure out how to
+fix the problem.
+
+If you use vcr logging you'll see these categories in your logs.
+
+### Serializers
+
+Serializers handle in what format cassettes are written to files on disk.
+The current options are YAML (default) and JSON. 
+
+An R6 class `Serializer` is the parent class for all serializer types;
+`YAML` and `JSON` are both R6 classes that inherit from `Serializer`. Both
+`YAML` and `JSON` define just two methods: `serialize()` and `deserialize()`
+for converting R structures to yaml or json, and converting yaml or json back
+to R structures, respectively.
+
+
+## Environments
+
+### Logging
+
+An internal environment (`vcr_log_env`) is used when you use logging.
+At this point it only keeps track of one variable - `file` - to be able
+to refer to what file is used for logging across many classes/functions
+that need to write to the log file.
+
+### A bit of housekeeping
+
+Another internal environment (`vcr__env`) is used to keep track of a
+few items, including the current cassette in use, and the last vcr error.
+
+### Lightswitch
+
+Another internal environment (`light_switch`) is used to keep track of users
+turning on and off `vcr`. See `?lightswitch`.
+
+
+
+## Footnotes
```

Comment:
given that `bookdown::bs4_book()` makes footnotes accessible on hover, and that pkgdown next version will do that as well, I think you should remove this section heading.

Reply:
will do